android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationBuilder.java

```diff
@@ -360,7 +360,7 @@ private AnimatorSet buildPropertyAnimators(int x, int y, int w, int h, int paren
 				for (Operation operation : operations) {
 					if (operation.anchorX != Ti2DMatrix.DEFAULT_ANCHOR_VALUE
 						|| operation.anchorY != Ti2DMatrix.DEFAULT_ANCHOR_VALUE) {
-						setAnchor(w, h, operation.anchorX, operation.anchorY);
+						setAnchor(w, h, anchorX, anchorY);
```

Comment:
I think the if-condition should be written like the below.
```java
final float EPSILON = Math.ulp(1.0f);
boolean hasAnchorXChanged = (Math.abs(operation.anchorX - anchorX) >= EPSILON);
boolean hasAnchorYChanged = (Math.abs(operation.anchorY - anchorY) >= EPSILON);
if (hasAnchorXChanged || hasAnchorYChanged) {
	setAnchor(w, h, anchorX, anchorY);
}
```

This way if the first time the operations were using `DEFAULT_ANCHOR_VALUE` and the anchor was changed to something else, then the operations would be updated instead of locking-in to using the previous values. The above effectively doing `operation.anchorX != anchorX`, but with floats.

_(I've never played with this code before. So, hopefully I'm understanding things correctly. Am I right?)_

Reply:
I don't think that's what was intended. The test case has an animation with anchor point that is different by the default one, which is then followed by an animation with the default one. And this will result in playing the first animation twice, because the difference will be less than EPSILON.

I also found out that we can have anchorPoint as a parameter in the creation dictionary of Ti.UI.2DMatrix. And my changes totally ignore that, which is not OK. So I will revisit the code. check what is the priority on iOS and fix it.